python/foglamp/services/core/api/notification.py

```diff
@@ -0,0 +1,524 @@
+        is_enabled = "true" if ((type(enabled) is str and enabled.lower() in ['true']) or (
+            (type(enabled) is bool and enabled is True))) else "false"
+
+        try:
+            # Get default config for rule and channel plugins
+            url = '{}/plugin'.format(request.url)
+            list_plugins = json.loads(await _hit_get_url(url))
+            rule_plugin_config = list_plugins['rules'][rule]
+            delivery_plugin_config = list_plugins['delivery'][channel]
+        except KeyError:
+            raise web.HTTPBadRequest(reason="Invalid rule plugin:[{}] and/or delivery plugin:[{}] supplied.".format(rule, channel))
+
+        # First create templates for notification and rule, channel plugins
+        post_url = 'http://{}:{}/foglamp/notification/{}'.format(_address, _port, urllib.parse.quote(name))
+        await _hit_post_url(post_url)  # Create Notification template
+        post_url = 'http://{}:{}/foglamp/notification/{}/rule/{}'.format(_address, _port, urllib.parse.quote(name), urllib.parse.quote(rule))
+        await _hit_post_url(post_url)  # Create Notification rule template
+        post_url = 'http://{}:{}/foglamp/notification/{}/delivery/{}'.format(_address, _port, urllib.parse.quote(name), urllib.parse.quote(channel))
+        await _hit_post_url(post_url)  # Create Notification delivery template
+
+        # Create configurations
+        storage = connect.get_storage_async()
+        config_mgr = ConfigurationManager(storage)
+        notification_config = {
+            "name": {
+                "description": "The name of this notification",
+                "type": "string",
+                "default": name,
+            },
+            "description": {
+                "description": "Description of this notification",
+                "type": "string",
+                "default": description,
+            },
+            "rule": {
+                "description": "Rule to evaluate",
+                "type": "string",
+                "default": rule,
+            },
+            "channel": {
+                "description": "Channel to send alert on",
+                "type": "string",
+                "default": channel,
+            },
+            "notification_type": {
+                "description": "Type of notification",
+                "type": "enumeration",
+                "options": NOTIFICATION_TYPE,
+                "default": notification_type,
+            },
+            "enable": {
+                "description": "Enabled",
+                "type": "boolean",
+                "default": is_enabled,
+            }
+        }
+        await _create_configurations(storage, config_mgr, name, notification_config,
+                                     rule, rule_plugin_config, rule_config,
+                                     channel, delivery_plugin_config, delivery_config)
+    except Exception as e:
+        raise web.HTTPBadRequest(reason=str(e))
```

Comment:
`HTTPInternalServerError` or add appropriate catch Exceptions with `HTTPBadRequest` in case of Key/Value Errors